src/backend/crew_ai/cleaned_llm_wrapper.py

```
@@ -45,6 +108,56 @@ def __init__(self, *args, **kwargs):
+                # Extract retry delay from error
+                error_str = str(e)
+                retry_delay = DynamicRateLimitHandler.extract_retry_delay(error_str)
+                
+                if retry_delay is None:
+                    # Default fallback if we can't parse the delay
+                    retry_delay = 60.0
+                    logger.warning(f"⚠️ Could not parse retryDelay, using default {retry_delay}s")
+                
+                logger.info(f"⏱️ Rate limit hit (attempt {attempt + 1}/{max_retries + 1}). "
+                           f"Waiting {retry_delay:.1f}s as specified by API...")
+                time.sleep(retry_delay)
```

Ruff (0.15.7)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)